src/crab/ebpf_domain.cpp

```diff
@@ -1561,6 +1562,28 @@ void ebpf_domain_t::operator()(const TypeConstraint& s) {
         require(m_inv, linear_constraint_t::FALSE(), "");
 }
 
+void ebpf_domain_t::operator()(const FuncConstraint& s) {
+    // Look up the helper function id.
+    const reg_pack_t& reg = reg_pack(s.reg);
+    auto src_interval = m_inv.eval_interval(reg.svalue);
+    if (std::optional<number_t> sn = src_interval.singleton()) {
```

Comment:
```suggestion
    if (auto sn = src_interval.singleton()) {
```
to avoid coercion when the expression is not actually an optional.